core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -166,6 +166,15 @@ class KryoSerializer(conf: SparkConf)
     kryo.register(Utils.classForName("scala.collection.immutable.$colon$colon"))
     kryo.register(classOf[ArrayBuffer[Any]])
 
+    // Add classes for Streaming
+    try {
+      kryo.register(
+        Utils.classForName("org.apache.spark.streaming.util.OpenHashMapBasedStateMap"),
```

Comment:
There is a tricky thing here.

Kryo assigns an unique id to each registered class and only writes its id, so the register order of classes should be same. Otherwise, the ids won't be matched and deserialization will fail.

However, for tests that start a local cluster, their executors have `OpenHashMapBasedStateMap` but the driver doesn't. So I added `OpenHashMapBasedStateMap` at the last class to make sure other classes's ids are same.